### PR TITLE
feature: add smtp settings

### LINKED ISSUE #24: feature request: be able to configure smtp settings
Being able to configure the SMTP settings so Zitadel is able to send E-Mails.

### PR DESCRIPTION
**Pull request by stebenz**
closes #24

### CODE REVIEW

File: zitadel/v2/smtp_config/funcs.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+	client, err := helper.GetAdminClient(clientinfo)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	resp, err := client.GetSMTPConfig(ctx, &admin.GetSMTPConfigRequest{})
+	if err != nil {
+		d.SetId("")
+		return nil
+		//return diag.Errorf("error while reading smtp config: %v", err)
```
Comment (eliobischof) [inline zitadel/v2/smtp_config/funcs.go:131]:
```suggestion
```

File: zitadel/v2/smtp_config/funcs.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+			SenderAddress: senderAddress,
+			SenderName:    senderName,
+			Tls:           tls,
+			Host:          host,
+			User:          user,
+		})
+		if err != nil {
+			return diag.Errorf("failed to update smtp config: %v", err)
+		}
+	} else {
```
Comment (eliobischof) [inline zitadel/v2/smtp_config/funcs.go:102]: if the password and another property is changed, terraform needs to run twice like this, right?

File: zitadel/v2/smtp_config/resource.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+				Description: "Address used to send emails.",
+			},
+			senderNameVar: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Sender name used to send emails.",
+			},
+			tlsVar: {
+				Type:        schema.TypeBool,
+				Required:    true,
```
Comment (eliobischof) [inline zitadel/v2/smtp_config/resource.go:23]:
```suggestion
				Required:    false,
```

File: zitadel/v2/smtp_config/resource.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+				Description: "TLS used to communicate with your SMTP server.",
+			},
+			hostVar: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Host address to your SMTP server.",
+			},
+			userVar: {
+				Type:        schema.TypeString,
+				Required:    true,
```
Comment (eliobischof) [inline zitadel/v2/smtp_config/resource.go:33]: is that really required?

File: zitadel/v2/smtp_config/resource.go [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+				Description: "User used to communicate with your SMTP server.",
+			},
+			passwordVar: {
+				Type:        schema.TypeString,
+				Required:    true,
```
Comment (eliobischof) [inline zitadel/v2/smtp_config/resource.go:38]: is that really required?

**Review by eliobischof — CHANGES_REQUESTED** [review]

File: zitadel/v2/smtp_config/resource.go (hunk 4 above)
Comment (stebenz) [reply zitadel/v2/smtp_config/resource.go:33]: In ZITADEL it's no required, but as you should be able to use an empty string it's easier to handle like that

File: zitadel/v2/smtp_config/resource.go (hunk 5 above)
Comment (stebenz) [reply zitadel/v2/smtp_config/resource.go:38]: In ZITADEL it's no required, but as you should be able to use an empty string it's easier to handle like that

File: zitadel/v2/smtp_config/resource.go (hunk 3 above)
Comment (stebenz) [reply zitadel/v2/smtp_config/resource.go:23]: In ZITADEL it's no required, but as the default would be false, I rather define it as required

File: zitadel/v2/smtp_config/resource.go (hunk 5 above)
Comment (eliobischof) [reply zitadel/v2/smtp_config/resource.go:38]: So, your idea is, that users should always specify all fields, without having the concept of defaults at all?

Comment (stebenz) [reply zitadel/v2/smtp_config/resource.go:38]: No? as there is no default definition for the password at all, I rather have a fully defined resource in terraform, as with only 5 attributes it has no impact. As SMTP settings are such a key resource.

**Review by eliobischof — APPROVED** [review]

**Pull request merged**